File: app/src/main/java/com/ubiqsmart/repository/data/WatchWallet.java
package com.ubiqsmart.repository.data;

import com.ubiqsmart.interfaces.StorableWallet;

import java.io.*;

public class WatchWallet implements StorableWallet, Serializable {

  private static final long serialVersionUID = -146500951598835658L;

  private String pubKey;
  private long dateAdded;

  public WatchWallet(String pubKey) {
    this.pubKey = pubKey.toLowerCase();
    this.dateAdded = System.currentTimeMillis();
  }

  public String getPubKey() {
    return pubKey;
  }

}

File: app/src/main/java/com/ubiqsmart/repository/data/WatchWallet.kt
package com.ubiqsmart.repository.data

import com.ubiqsmart.interfaces.StorableWallet
import java.io.Serializable

data class WatchWallet(override val pubKey: String) : StorableWallet, Serializable {

    val dateAdded: Long = System.currentTimeMillis()

}
